extensions/gulp/src/main.ts

```diff
@@ -87,6 +88,11 @@ class FolderDetector {
 		this.fileWatcher.onDidChange(() => this.promise = undefined);
 		this.fileWatcher.onDidCreate(() => this.promise = undefined);
 		this.fileWatcher.onDidDelete(() => this.promise = undefined);
+		let nodeModulesPattern = path.join(this._workspaceFolder.uri.fsPath, 'node_modules');
```

Comment:
Could we not have one matcher like this?

let pattern = path.join(this._workspaceFolder.uri.fsPath, '{node_modules,gulpfile{.babel.js,.js,.ts}}')

Reply:
That's definitely better, I didn't know that the patter could have multiple folders!